src/sctools/bam.py

```diff
@@ -464,7 +493,26 @@ def __repr__(self) -> str:
         return 'query_name'
 
 
-class CellMoleculeGeneQueryNameSortOrder(AlignmentSortOrder):
+class TagsQueryNameSortOrder(AlignmentSortOrder):
+    """Hierarchical alignment record sort order (tag1 >= tag2 >= ... >= query name)."""
+    def __init__(self, tags) -> None:
+        assert tags, "tags"
+        self.tags = tags
+
+    def _get_sort_key(self, alignment: pysam.AlignedSegment):
+        keys = [get_tag_or_default(alignment, tag, default='N') for tag in self.tags]
+        keys.append(alignment.query_name)
+        return keys
+
+    @property
+    def key_generator(self):
+        return self._get_sort_key
+
+    def __repr__(self) -> str:
+        return 'hierarchical__tags_query_name'
+
+
+class CellMoleculeGeneQueryNameSortOrder(TagsQueryNameSortOrder):
```

Comment:
Is this class needed anymore now that there is a more generalized `TagsQueryNameSortOrder` class? It looks like it's only being used in some tests

Reply:
@ambrosejcarr @mbabadi if you want to keep the `CellMoleculeGeneQueryNameSortOrder` class for any reason let me know. Otherwise I will take it out, now that the more generic `TagQueryNameSortOrder` exists.

Reply:
I'm fine with taking it out.